Src/DependencyCollector/Net46.Tests/DependencyTrackingTelemetryModuleTestNet46.cs

```diff
@@ -117,28 +129,42 @@ public void TestDependencyCollectionEventSource404()
                 module.Initialize(this.config);
                 Assert.IsFalse(DependencyTableStore.IsDesktopHttpDiagnosticSourceActivated);
 
-                var url = new Uri("http://google.com/404");
                 HttpClient httpClient = new HttpClient();
-                httpClient.GetStringAsync(url).ContinueWith(t => { }).Wait();
+                using (new LocalServer(
+                    LocalhostUrl, 
+                    ctx => 
+                    {
+                        // https://github.com/Microsoft/ApplicationInsights-dotnet-server/issues/548
+                        // for quick unsuccesfull response OnEnd is fired too fast after begin (before it's completed)
+                        // first begin may take a while because of lazy initializations and jit compiling
+                        // let's wait a bit here.
+                        Thread.Sleep(20);
+                        ctx.Response.StatusCode = 404;
+                    }))
+                {
+                    await httpClient.GetAsync(LocalhostUrl).ContinueWith(t => { });
+                }
 
-                this.ValidateTelemetryForEventSource(this.sentTelemetry.Single(), url, null, false, "404");
+                this.ValidateTelemetryForEventSource(this.sentTelemetry.Single(), new Uri(LocalhostUrl), null, false, "404");
             }
         }
 
         [TestMethod]
         [Timeout(5000)]
-        public void TestDependencyCollectionDiagnosticSource404()
+        public async Task TestDependencyCollectionDiagnosticSource404()
         {
             using (var module = new DependencyTrackingTelemetryModule())
             {
                 module.ProfileQueryEndpoint = FakeProfileApiEndpoint;
                 module.Initialize(this.config);
 
-                var url = new Uri("http://google.com/404");
                 HttpClient httpClient = new HttpClient();
-                httpClient.GetStringAsync(url).ContinueWith(t => { }).Wait();
+                using (new LocalServer(LocalhostUrl, ctx => ctx.Response.StatusCode = 404))
+                {
+                    await httpClient.GetStringAsync(LocalhostUrl).ContinueWith(t => { });
+                }
 
-                this.ValidateTelemetryForDiagnosticSource(this.sentTelemetry.Single(), url, null, false, "404");
+                this.ValidateTelemetryForDiagnosticSource(this.sentTelemetry.Single(), new Uri(LocalhostUrl), null, false, "404");
```

Comment:
can you add a comment to make it explicit that DiagnosticSource alone is not collecting dependency here, but a combined effort by DiagnosticSource and FrameworkES, but the expected prefix will be 'rdddsd' because we mark the prefix based on who started monitoring the dependency.  (This avoids any confusion to future readers about ability of DiagnosticSource to collect failures)